test/Microsoft.IdentityModel.Tokens.Tests/Validation/LifetimeValidationResultTests.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.TestUtils;
+using Xunit;
+
+namespace Microsoft.IdentityModel.Tokens.Validation.Tests
+{
+    public class LifetimeValidationResultTests
+    {
+        [Theory, MemberData(nameof(ValidateLifetimeTestCases), DisableDiscoveryEnumeration = true)]
+        public void ValidateLifetime(ValidateLifetimeTheoryData theoryData)
+        {
+            CompareContext context = TestUtilities.WriteHeader($"{this}.LifetimeValidatorTests", theoryData);
+
+            LifetimeValidationResult lifetimeValidationResult = Validators.ValidateLifetime(
+                theoryData.NotBefore,
+                theoryData.Expires,
+                theoryData.SecurityToken,
+                theoryData.ValidationParameters,
+                new CallContext());
+
+            if (lifetimeValidationResult.Exception == null)
+                theoryData.ExpectedException.ProcessNoException();
+            else
+                theoryData.ExpectedException.ProcessException(lifetimeValidationResult.Exception, context);
+
+            IdentityComparer.AreLifetimeValidationResultsEqual(
+                lifetimeValidationResult,
+                theoryData.LifetimeValidationResult,
+                context);
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<ValidateLifetimeTheoryData> ValidateLifetimeTestCases
+        {
+            get
+            {
+                DateTime now = DateTime.UtcNow;
+                DateTime oneHourFromNow = EpochTime.DateTime(EpochTime.GetIntDate(DateTime.UtcNow + TimeSpan.FromHours(1)));
```

Comment:
You don't need to change this, but more of a heads up.

There are slightly easier to read Apis for DateTime, for example

```cs
DateTime.UtcNow.AddHours(1);
```

Reply:
Sorry I'm missing something here,

Why did we go DateTime -> long -> DateTime?

Reply:
These were written by Copilot. I just named the variable and trusted it to come up with a good way to add the time to the DateTime object. I can change to the format you wrote above.

Reply:
Code was updated.